src/event/schema/request/event.schema.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from "zod";
+import mongoose from "mongoose";
+
+export const idSchema = z.string().refine((id) => mongoose.Types.ObjectId.isValid(id), {
+  message: "Invalid ObjectId",
+});
+
+export const eventSchema = z.object({
+  _id: z.instanceof(mongoose.Types.ObjectId).optional(),
+  name: z.string().min(1, "Event name is required"),
+  description: z.string().optional(),
+  date: z.date().optional(),
```

Comment:
Tady bacha, nejsem si jistý, že ti to automaticky zkonvertuje `string` poslaný v jsonu do `date` objektu. Lepší by bylo tu použít `z.coerce.date`